coriolis/tests/conductor/rpc/test_server.py

```diff
@@ -18,6 +22,345 @@ def setUp(self):
         super(ConductorServerEndpointTestCase, self).setUp()
         self.server = server.ConductorServerEndpoint()
 
+    @mock.patch.object(rpc_worker_client.WorkerClient, 'from_service_definition')
+    @mock.patch.object(server.ConductorServerEndpoint, '_scheduler_client')
+    def test_get_all_diagnostics(self, mock_scheduler_client, mock_from_service_definition):
+        mock_scheduler_client.get_workers_for_specs.side_effect = Exception()
+        self.assertRaises(Exception, lambda: self.server.get_all_diagnostics(
+            mock.sentinel.context))
+        mock_scheduler_client.get_workers_for_specs.side_effect = None
+
+        diagnostics = self.server.get_all_diagnostics(mock.sentinel.context)
+        # last element is the one from the scheduler client
```

Comment:
I'd recommend trying to avoid relying on index ordering (or any other such "implementation details") in tests, as this test could yield a semi-false negative if we ever decide to add a new service at the end of the diagnostics lists.

Maybe try filtering the service definitions via [RabbitMQ topic](https://github.com/cloudbase/coriolis/blob/master/coriolis/constants.py#L337) or somm?